lib/bot/trello_helper.rb

```diff
@@ -0,0 +1,211 @@
+    # - Checked In Progress Bugs: Move status to POST
+    # - Checked Complete Bugs: Ignored, no further states to process
+    def process_checked_bugs
+      bugs_by_status = Hash.new([])
+
+      # find all the "checked" bugs for each status across all the teams
+      trello_team_names.each do |team|
+        %w(ASSIGNED ON_DEV).each do |bug_status|
+          bug_card = bug_card_for(team, bug_status)
+          bug_card.checklists.each do |checklist|
+            bug_ids = checklist.items.select { |i| i.state == "complete" }.collect { |i| bug_id_from_checklist_item(i) }
+            bugs_by_status[bug_status].concat(bug_ids)
+          end
+        end
+      end
+
+      # batch update all the bugs for each status
+      bugs_by_status.each do |current_status, bug_ids|
+        new_status = case current_status
+          when "ASSIGNED"; then "ON_DEV"
+          when "ON_DEV";   then "POST"
+        end
+        bugzilla::Base.service.update(bug_ids, :status => new_status)
+      end
+    end
+
+    # Looks up the bugs that should be on each bug card and entirely replaces
+    # the checklists on those cards with the found bugs.
+    #
+    def update_bug_checklists
+      Settings.trello.teams.each do |team, team_data|
+        bugs  = bugs_for(team)
+        board = orgnization.board[team_data.board]
+        bugs.each do |status, bugs_by_assigned|
+          card = bug_card_for(team, status)
+          card.remove_all_checklists
+
+          bugs_by_assigned.each do |assigned, assigned_bugs|
+            checklist = Trello::Checklist.create(:name => assigned, :board_id => board.id)
+            checklist_items = assigned_bugs.collect { |bug| format_bug_checklist_item(bug) }
+            checklist.add_items(checklist_items)
+          end
+        end
+      end
+    end
+
+    # Returns a formatted string that can be used as a checklist item name for a
+    # bug
+    def format_bug_checklist_item(bug)
+      priority = %w(urgent high).include?(bug.priority) ? "**#{bug.priority}**" : bug.priority
+      "[[#{bug.id}] #{bug.summary} (#{priority})](#{bz_url(bug)})"
+    end
+
+    def bug_id_from_checklist_item(checklist_item)
+      checklist_item.name.match(/show_bug\.cgi\?id=([0-9]*)/) { |match| match[1] }
+    end
+
+    def bz_url(bug)
+      "https://bugzilla.redhat.com/show_bug.cgi?id=#{bug.id}"
+    end
+
+    def organization
+      @organization ||= Trello::Organization.find(Settings.trello.organization_id)
+    end
+
+    private
+
+    def bugzilla
+      begin
+        # this access will force a name error if the service hasn't been set yet
+        ActiveBugzilla::Base.service
+      rescue NameError
+        service = ActiveBugzilla::Service.new(
+          Settings.bugzilla_credentials.bugzilla_uri,
+          Settings.bugzilla_credentials.username,
+          Settings.bugzilla_credentials.password
+        )
+        ActiveBugzilla::Base.service = service
+      end
+      ActiveBugzilla
```

Comment:
K, I didn't dig very deep into the other helpers to find this ... I can replace this with a reference to another helper.